webgpu.h

```diff
@@ -433,12 +434,17 @@ typedef enum WGPUTextureUsage {
 typedef WGPUFlags WGPUTextureUsageFlags;
 
 
+typedef struct WGPUChainedStruct {
+    struct WGPUChainedStruct const * nextInChain;
+    WGPUSType sType;
+} WGPUChainedStruct;
+
 typedef struct WGPUAdapterDescriptor {
-    void const * nextInChain;
+    WGPUChainedStruct const * nextInChain;
```

Comment:
The `struct` should only be needed in `WGPUChainedStruct` because at that point the `typedef` for `WGPUChainedStruct` hasn't taken effect yet. Other places should see the `typedef` correctly.

According to this [godbolt](https://godbolt.org/z/BaKUfq) it compiles in C.